.github/workflows/tests.yml

```diff
@@ -0,0 +1,11 @@
+on: [pull_request]
```

Comment:
This should include pushing the branch too.

Reply:
I was wondering about that. Am I missing something?

> We only need to trigger our tests on 'pull_request' events,
which includes a 'synchronize' activity for any changes.

Reply:
When a PR is open, and new commits are pushed to a branch that has an open PR, Actions will trigger based on the `pull_request` "synchronise" event.

I agree with Kevin that it's useful to preserve the behaviour we get from Jenkins of also triggering on `push` events to branches that don't have PRs, otherwise people will have to open PRs too early (and deal with the world seeing our force-pushes 😬) to check if our tests pass cleanly.

You can exclude branches, so we can _not_ run this test action on pushes to `master`.

Reply:
It's a common part of my workflow to have CI run before I open a PR. I had assumed that most people did the same too?

Reply:
Ah, I see. That was a nice-to-have for me. We shouldn't say `on: [pull_request, push]`, since that will cause the tests to run twice, as both trigger the action.

I *think* we just need the `push` trigger. Also updated (and tested) the commit so that we ensure the master branch actually builds before we release anything.

@issyl0 @kevindew could you take another look?

Reply:
There's background and further discussion on this in #govuk-dev-tools and https://github.com/alphagov/govuk-rfcs/pull/123/commits/efc476db36a9d47e25e65d9d898ab9cce8a06925.